python/pyarrow/_dataset.pyx

```diff
@@ -467,12 +467,13 @@ cdef class FileSystemDataset(Dataset):
     cdef:
         CFileSystemDataset* filesystem_dataset
 
-    def __init__(self, fragments, Schema schema, FileFormat format,
+    def __init__(self, filesystem, fragments, Schema schema, FileFormat format,
```

Comment:
Can you add this to the docstring? 
I would also move this keyword after `format`, I think fragments is most logical to come first. 

If we want to have this backwards compatible, we could also make this keyword optional, and if not specified, take it from the first fragment.